Source/WebCore/rendering/style/StyleImageSet.cpp

```diff
@@ -90,14 +90,16 @@ ImageWithScale StyleImageSet::bestImageForScaleFactor()
         const auto& image = m_images[index];
         if (!image.mimeType.isNull() && !MIMETypeRegistry::isSupportedImageMIMEType(image.mimeType))
             continue;
-
+        if (!result.image->isInvalidImage() && result.scaleFactor == image.scaleFactor)
```

Comment:
I guess this works when the identical resolutions are not next to each other because we use sorted indices?

Reply:
Ah yes, I should read the commit message more carefully, nice